wdl2cwl/main.py

```diff
@@ -236,16 +236,20 @@ def get_literal_value(expr: WDL.Expr.Base) -> Optional[Any]:
   if (contents == 'true') { return true;}
   if (contents == 'false') { return false;}
   throw "'read_boolean' received neither 'true' nor 'false': " + self[0].contents;
+}""",
+    "read_lines": r"""${
+  var contents = self[0].contents.replace(/\r\n$/, "").replace(/\n$/, "").replace(/\r$/, "");
```

Comment:
Looks good to me. Do you know what's the `self` variable above? That's JS right? So not the Python `self`... I couldn't find it looking around this line, so more for my curiosity :) 

Thanks!

Reply:
https://www.commonwl.org/v1.2/CommandLineTool.html#Parameter_references

>  The following parameters must be provided:
>
>    `inputs`: The input object to the current Process.
>    `self`: A context-specific value. The contextual values for `self` are documented for specific fields elsewhere in this specification. If a contextual value of 'self' is not documented for a field, it must be 'null'.
>    `runtime`: An object containing configuration details. Specific to the process type. An implementation may provide opaque strings for any or all fields of runtime. These must be filled in by the platform after processing the Tool but before actual execution. Parameter references and expressions may only use the literal string value of the field and must not perform computation on the contents, except where noted otherwise.

These particular expressions end up in the `outputEval` field, so we look at https://www.commonwl.org/v1.2/CommandLineTool.html#CommandOutputBinding to learn that

> If `glob` was specified, the value of `self` must be an array containing file objects that were matched. If no files were matched, `self` must be a zero length array; if a single file was matched, the value of `self` is an array of a single element.

Reply:
Thanks!!